### ISSUE TITLE
One simple question about zk_trie_impl

### ISSUE THREAD

**Issue by hsyodyssey** [scroll-tech/zktrie#1]
Hi, may I ask what the purpose of this [line](https://github.com/scroll-tech/zktrie/blob/main/trie/zk_trie_impl.go#L119) in `TryUpdate` function is?

The value of `dbKeyRootNode ` is not changed in `TryUpdate` process . So this function call will always insert the KV with the same `Key` into the DB, right?

Please correct me if I have a misunderstanding.

**Comment by shu-unifra** [comment]
The value of dbKeyRootNode is (DBEntryTypeRoot+mt.rootKey) and mt.rootKey is changed in TryUpdate,  so......

**Comment by hsyodyssey** [comment]
> The value of dbKeyRootNode is (DBEntryTypeRoot+mt.rootKey) and mt.rootKey is changed in TryUpdate, so......

Yeah, but the `Key` of dbKeyRootNode is [constant](https://github.com/scroll-tech/zktrie/blob/11cc53f2d799d4050baf3a0a3d1b4e4d1f84a606/trie/zk_trie_impl.go#L533) (`dbKeyRootNode = []byte("currentroot")`), right? The function directly insert KVs with the same `Key` (sure different `Value`) to underlying database. 

My question is, if more than one zkTries exist at the same time, will there be conflict or overwriting in their Root updates?

**Comment by shu-unifra** [comment]
> > The value of dbKeyRootNode is (DBEntryTypeRoot+mt.rootKey) and mt.rootKey is changed in TryUpdate, so......
> 
> Yeah, but the `Key` of dbKeyRootNode is [constant](https://github.com/scroll-tech/zktrie/blob/11cc53f2d799d4050baf3a0a3d1b4e4d1f84a606/trie/zk_trie_impl.go#L533) (`dbKeyRootNode = []byte("currentroot")`), right? The function directly insert KVs with the same `Key` (sure different `Value`) to underlying database.
> 
> My question is, if more than one zkTries exist at the same time, will there be conflict or overwriting in their Root updates?

OK, I got your question, but I don't know why~

**Comment by hsyodyssey** [comment]
Haha, there seems to be no conflict problem encountered now. So I'm curious if there are other mechanisms to avoid read conflicts.

**Comment by hsyodyssey** [comment]
Having understood that, I think I know why there is no conflict at the current codebase.

**Comment by shu-unifra** [comment]
> Having understood that, I think I know why there is no conflict at the current codebase.

Can you tell me about it please?

**Comment by hsyodyssey** [comment]
> > Having understood that, I think I know why there is no conflict at the current codebase.
> 
> Can you tell me about it please?

Basically, I think this function call is redundant since it just inserts a KV with a constant key. There is no other function to query this KV and the Root value of each contract's Storage Trie does not point to this KV. Therefore, this function call is just writing continuously and does not cause any conflicts. Maybe could optimize this area in the future 😊.
